android/src/main/kotlin/com/hypertrack/sdk/flutter/common/HyperTrackSdkWrapper.kt

```diff
@@ -0,0 +1,246 @@
+    onInstanceCall: (sdk: HyperTrack) -> T
+  ): Result<T> {
+    return this.sdkInstance.let { instance ->
+      if (instance == null) {
+        Failure(Exception("HyperTrack SDK is not initialized"))
+      } else {
+        try {
+          Success(onInstanceCall(instance))
+        } catch (e: Exception) {
+          Failure(e)
+        }
+      }
+    }
+  }
+
+  private fun getLocationError(error: OutageReason): LocationError {
+    val blockersErrors = getHyperTrackErrorsFromBlockers()
+    return when (error) {
+      OutageReason.NO_GPS_SIGNAL -> {
+        Errors(setOf(HyperTrackError.gpsSignalLost) + blockersErrors)
+      }
+      OutageReason.MISSING_LOCATION_PERMISSION -> {
+        Errors(setOf(HyperTrackError.locationPermissionsDenied) + blockersErrors)
+      }
+      OutageReason.LOCATION_SERVICE_DISABLED -> {
+        Errors(setOf(HyperTrackError.locationServicesDisabled) + blockersErrors)
+      }
+      OutageReason.MISSING_ACTIVITY_PERMISSION -> {
+        Errors(setOf(HyperTrackError.motionActivityPermissionsDenied) + blockersErrors)
+      }
+      OutageReason.NOT_TRACKING -> {
+        NotRunning
+      }
+      OutageReason.START_HAS_NOT_FINISHED -> {
+        Starting
+      }
+      OutageReason.RESTART_REQUIRED -> {
+        throw IllegalStateException("RESTART_REQUIRED must not be returned")
+      }
+    }
+  }
+
+  private fun getLocationError(error: GeotagResult.Error.Reason): LocationError {
+    val blockersErrors = getHyperTrackErrorsFromBlockers()
+    return when (error) {
+      GeotagResult.Error.Reason.NO_GPS_SIGNAL -> {
+        Errors(setOf(HyperTrackError.gpsSignalLost) + blockersErrors)
+      }
+      GeotagResult.Error.Reason.MISSING_LOCATION_PERMISSION -> {
+        Errors(setOf(HyperTrackError.locationPermissionsDenied) + blockersErrors)
+      }
+      GeotagResult.Error.Reason.LOCATION_SERVICE_DISABLED -> {
+        Errors(setOf(HyperTrackError.locationServicesDisabled) + blockersErrors)
+      }
+      GeotagResult.Error.Reason.MISSING_ACTIVITY_PERMISSION -> {
+        Errors(setOf(HyperTrackError.motionActivityPermissionsDenied) + blockersErrors)
+      }
+      GeotagResult.Error.Reason.NOT_TRACKING -> {
+        NotRunning
+      }
+      GeotagResult.Error.Reason.START_HAS_NOT_FINISHED -> {
+        Starting
+      }
+    }
+  }
+
+  private fun getHyperTrackErrorsFromBlockers(): Set<HyperTrackError> {
+    return HyperTrack.getBlockers().map {
+      when (it) {
+        Blocker.LOCATION_PERMISSION_DENIED -> {
+          HyperTrackError.locationPermissionsDenied
+        }
+        Blocker.LOCATION_SERVICE_DISABLED -> {
+          HyperTrackError.locationServicesDisabled
+        }
+        Blocker.ACTIVITY_PERMISSION_DENIED -> {
+          HyperTrackError.motionActivityPermissionsDenied
+        }
+        Blocker.BACKGROUND_LOCATION_DENIED -> {
+          HyperTrackError.locationPermissionsInsufficientForBackground
+        }
+      }
+    }.toSet()
+  }
+
+}
```

Comment:
this file has a different format than others, 2 spaces as 1 intend instead of 4

Reply:
ANd not only this .kt file, actually all files below have a different format

Reply:
IDE autoformat is working weird for Flutter plugins for some reason (probably it uses Flutter settings). I would not try fixing it right now.

Reply:
fixed